v23/security/access/access.vdl.go

```diff
@@ -479,7 +479,6 @@ var (
 	__VDLType_string_4 *vdl.Type
```

Comment:
Why are so many of these variables still declared and initialized?

Reply:
They're still initialized in `__VDLInit()`, and used by other parts of the code, e.g. in various `VDLWrite` methods, which are used by `vom`.

The specific problem we encountered with the caveats is that they are constants which contain vdl `TypeObject`s.  I.e. we allow creating constant (values) in `vdl` that contain instances of types, so that you can express things like "my caveat has a parameter type `struct{Foo bool; Bar string}`".  And it turns out we had a long-standing bug in the codegen for vdl constants that contain TypeObjects, that was only exposed by Go 1.13, wrt init ordering.

We don't think the other uses of `TypeObject` suffer from the same init ordering issue, because all of those uses are safely within functions, which are never called by initializers or `init` funcs.  Also note that `vdl.TypeOf` is somewhat expensive, since it makes heavy use of reflection, and the whole reason we added these `__VDLType_*` variables was to improve `vom` performance, since it needs the types to perform encoding and decoding.  So this seemed like the simplest solution.